src/fsharp/TypeChecker.fs

```diff
@@ -699,6 +699,36 @@ let ImplicitlyOpenOwnNamespace tcSink g amap scopem enclosingNamespacePath env =
 // Helpers for unification
 //------------------------------------------------------------------------- 
 
+/// When the context is matching the oldRange then this function shrinks it to newRange.
+/// This can be used to change context over no-op expressions like parens.
+let ShrinkContext env oldRange newRange =
+    match env.eContextInfo with
+    | ContextInfo.NoContext
+    | ContextInfo.RecordFields
+    | ContextInfo.TupleInRecordFields
+    | ContextInfo.ReturnInComputationExpression
+    | ContextInfo.YieldInComputationExpression
+    | ContextInfo.RuntimeTypeTest _
+    | ContextInfo.DowncastUsedInsteadOfUpcast _ ->
+        env
+    | ContextInfo.CollectionElement (b,m) ->
+        if m <> oldRange then env else
```

Comment:
Maybe make this `if` a `when` in the pattern?

Reply:
I want to have it being a total function so that it breaks when we add a new ContextInfo

Reply:
I guess that makes sense, although | _ -> env would certainly eliminate a bunch of redundant typing and allow the pattern match to do work only when you need to make the contextinfo with newrange.  I can easily see someone refactoring this in a PR at some point in the future ... and no one batting an eye.

Reply:
I really really think this one needs to be total so that future self gets hint by compiler about missing pattern matching